ibis/expr/operations/logical.py

```diff
@@ -164,37 +168,130 @@ class Where(Value):
     output_shape = rlz.shape_like("bool_expr")
 
 
-class _AnyBase(Value):
-    # Depending on the kind of input boolean array, the result might either be
-    # array-like (an existence-type predicate) or scalar (a reduction)
+class Negatable:
+    @property
+    @abc.abstractmethod
+    def _negated_class(self) -> type:  # pragma: no cover
+        ...
+
+    def negate(self):
+        return self._negated_class(*self.args)
+
+
+@public
+class Any(Reduction, Negatable):
```

Comment:
Move these classes to `reduction.py` and keep aliases in this file using `public()`?

Reply:
Done.